server/routes/api/__test__/auth.unit.test.js

```diff
@@ -0,0 +1,48 @@
+const request = require('supertest');
+const app = require('../../../../app');
+const User = require('../../../models/user');
+
+describe('POST Auth/Login', () => {
+  beforeAll(() => {
+    return User.create({ email: 'testUser@test.com', password: 'test' });
+  });
+
+  afterAll(() => {
+    return User.findOneAndDelete({ email: 'testUser@test.com' });
+  });
+
+  it('will return error with invalid email', async () => {
+    const res = await request(app)
+      .post('/api/auth/login')
+      .type('form')
+      .send('email=test@test'); // send post with invalid invalid email
+    expect(res.statusCode).toEqual(400);
+    expect(res.body.error.message).toBe(
+      'Incorrect email and password combination.'
+    );
+  });
+
+  it('will return error with invalid password', async () => {
+    const res = await request(app)
+      .post('/api/auth/login')
+      .type('form')
+      .send('email=test@test.com')
+      .send('password=wrongpassword'); // send post with invalid password
+    expect(res.statusCode).toEqual(400);
+    expect(res.body.error.message).toBe(
+      'Incorrect email and password combination.'
+    );
+  });
+
+  it('will return user object on successful login', async () => {
+    const res = await request(app)
+      .post('/api/auth/login')
+      .type('form')
+      .send('email=testUser@test.com')
+      .send('password=test'); // send post with valid email and password
+    expect(res.statusCode).toEqual(200);
+    expect(Object.keys(res.body)).toEqual(
```

Comment:
According to line 37, you want this test to check and see if a user object comes back (or in other words, whether or not res.body looks like it has the information matching the user you logged in as). Instead, it looks like you're only checking to see whether or not the user object is coming back without the password attached. 

In addition to the test you wrote, you might also want to see if the object being returned is actually a user object, along with whether or not it's the right user.

Reply:
Makes sense. I changed the test to expect the keys from the user object, not including the password, as well as checking for correct user email.